### DISCUSSION TITLE
How to make count up in flutter/dart like the digit count up in aviator game

### DISCUSSION THREAD

**Discussion by king2636** [community/community#124622] [Programming Help]
### Body

I want to a make an auto count up in flutter, the count up starts automatically and it can stop at anytime giving a random digit value everytime it starts again. Just like the count up in aviator games. Please can anyone help.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by davevad93** [comment]
Hi @king2636 ,

I'm not sure if that's exactly what you're looking for, but here's an example of a `Dart` implementation of it. You can check how it works using an online compiler with an integrated terminal such as [Online Dart Compiler](https://www.tutorialspoint.com/execute_dart_online.php):

```Dart
import 'dart:async';
import 'dart:io';
import 'dart:math';

void main() {
  Counter counter = Counter();
  counter.startCounter();

  // Event listener for stopping and restarting the counter when pressing a key 
  stdin.listen((_) {
    counter.stopCounter();
    counter.startCounter();
  });
}

class Counter {
  int _counter = 0; 
  Timer? _timer; 
  final Random _random = Random(); 

  void startCounter() {
    // Set the counter to a random value between 0 and 999 (you can change it to whatever value you prefer)
    _counter = _random.nextInt(1000);  
    _timer = Timer.periodic(Duration(milliseconds: 100), (timer) {
      _counter++;
      print('Counter: $_counter');
    });
  }

  void stopCounter() {
    if (_timer != null && _timer!.isActive) {
      _timer?.cancel();
      // Stops the counter and prints the value when pressing a key
      print('Stopped at: $_counter');
    }
  }
}
```

**Reply by king2636** [reply]
![011](https://github.com/community/community/assets/103449452/507a09e9-35ca-49ab-bca9-0555fc98acca)

I want the counter to be like that one in the sample above
